streams-contrib/streams-provider-moreover/src/main/java/org/apache/streams/data/moreover/MoreoverProvider.java

```diff
@@ -93,17 +109,19 @@ public StreamsResultSet readRange(DateTime start, DateTime end) {
 
     @Override
     public boolean isRunning() {
-        return !executor.isShutdown() && !executor.isTerminated();
+        return !executor.isTerminated() || !this.providerQueue.isEmpty();
```

Comment:
All tasks will be submitted at the startStream call.  At that point shutdown will be called, because no new task should be added to executor.  So it should always shutdown.  it is finished running when all threads have terminated (hence the executor is terminated) and when all data has been removed from its queue.